File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        stack = []
        n = 0 #number of visited elements
        current = root

        while stack or current:
            while current:
                stack.append(current) #add the visited node
                current = current.left #and move the pointer to  the left node (with smaller values)

            #after all smaller values are added (current becomes null)
            current = stack.pop()
            n += 1
            if n == k:
                return current.val   
            current = current.right



        

File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int current;

        for (int i = 0; i < nums.size();)
        {
            for (int j = i + 1; j < nums.size();)
            {
                current = nums[i];
                if (current == nums[j])
                {
                    return true;
                }
                j++;
            }
            i++;
        }

        return false;
    }
};
